library/psa_crypto.c

```diff
@@ -7007,14 +7007,14 @@ static psa_status_t psa_validate_key_type_and_size_for_key_generation(
             return PSA_ERROR_NOT_SUPPORTED;
         }
     } else
-#endif /* defined(PSA_WANT_KEY_TYPE_RSA_KEY_PAIR) */
+#endif /* defined(MBEDTLS_PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_LEGACY) */
 
-#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR)
+#if defined(MBEDTLS_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_LEGACY)
     if (PSA_KEY_TYPE_IS_ECC(type) && PSA_KEY_TYPE_IS_KEY_PAIR(type)) {
         /* To avoid empty block, return successfully here. */
         return PSA_SUCCESS;
     } else
-#endif /* defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) */
+#endif /* defined(MBEDTLS_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_LEGACY) */
 
 #if defined(PSA_WANT_KEY_TYPE_DH_KEY_PAIR)
```

Comment:
> So maybe we don't need to provide the legacy symbol at all?

I'm sure @mpg will provide a more detailed answer on this, but in the meanwhile I can tell you what we agreed on a private Slack discussion. In order not to make this PR too big this PR should: 

- deprecate `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR` and `MBEDTLS_PSA_ACCEL_KEY_TYPE_xxx_KEY_PAIR`
- replacing them with library's code and tests' dependencies with their `_LEGACY` temporary version
- provide new ones (with `_USE, _IMPORT, _EXPORT, _DERIVE, _GENERATE`)

Then later there will be another PR which will aim at removing those `_LEGACY` internal symbols with the proper `_USE, _IMPORT, _EXPORT, _DERIVE, _GENERATE` in library's code and tests.

Reply:
> I meant maybe we don't need to keep the legacy public symbols for DH, since they were never released.

Yes, that's what I meant with the last item in the issue's description: "Remove PSA_WANT_KEY_TYPE_DH_KEY_PAIR (it wasn't in any released version so far)." Now, we might choose to use an internal `_LEGACY` symbol to replace it for now, but we want the public one gone ASAP, while we still can.